diff --git a/CMakeLists.txt b/CMakeLists.txt
index e890133..3e7689f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,7 @@ find_package(GSL)
 if (GSL_FOUND)
     message("Found GSL library ${GSL_INCLUDE_DIR}, will build MUSIC with GSL")
     include_directories(${GSL_INCLUDE_DIR})
+    set (ENV{CPATH} ${GSL_INCLUDE_DIR})
 elseif()
     message("Did not find GSL library, will build MUSIC alone")
 endif()
